File: 792-binary-search/binary-search.java
class Solution {
    public static int binsrch(int [] nums, int target, int left, int right){
        int mid = left + (right - left) / 2;
        // base case 1
        if (left > right){ 
            return -1;
        }
        // base case 2
        if (nums[mid] == target){
            return mid;
        }
        // recursive case 
        else {
            if (nums[mid] < target){
                return binsrch(nums, target, mid + 1, right);
            }
            else{
                return binsrch(nums, target, left, mid - 1);
            }
        }
    }

    public int search(int[] nums, int target) {
        int left = 0;
        int right = nums.length - 1;
        
        int ans = binsrch(nums, target, left, right);
        return ans;
    }
}

File: 792-binary-search/binary-search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        left = 0
        right = len(nums)-1

        def binsrch(nums, left, right, target):
            mid = left + (right - left) // 2
            if left > right: # base case 1
                return -1

            if nums[mid] == target: # base case 2
                return mid
            else: # recursive  case
                if nums[mid] < target: # rc1
                    return binsrch(nums, mid + 1, right, target)
                else: # rc2
                    return binsrch(nums, left, mid - 1, target)
        ans = binsrch(nums, left, right, target)
        return ans
        